src/xdr/Stellar-ledger.x

```diff
@@ -84,6 +106,8 @@ struct LedgerHeader
     {
     case 0:
         void;
+    case 1:
```

Comment:
Done

Reply:
That's a good point. This change would help though if more upgrade options are added in the future.

Reply:
Agreed, I'm not opposed to it; I just wanted to understand. You've wrapped quite a bit of code with exception handling, would you want to do the same thing for any bad message size exception in that block?

Reply:
Yeah. This will just result in a (somewhat) meaningful failure during catchup, instead of crashing with no information. I could just wrap the `readOne` call instead, but I don't think it really matters here.

Reply:
minor: this issue is still present in simulation functions, https://github.com/stellar/stellar-core/blob/e33b6f0150941fa84483b964e6c44cfdcffce4e7/src/main/CommandLine.cpp#L1524 and https://github.com/stellar/stellar-core/blob/bdb165b75613bf23e638dd32f8a6c9c07f26791f/src/catchup/simulation/TxSimApplyTransactionsWork.cpp#L399

I don't think it's a huge issue, but it would probably be better to provide more context via logs and exit gracefully instead of crashing

Reply:
also, I think it would be cleaner to wrap just the `readOne` call in a try/catch block, I don't think we expect this exception in the rest of this function?

Reply:
Fixed the simulation issues and wrapped just `readOne`.